com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/ExtractStatusCallback.java

```diff
@@ -0,0 +1,48 @@
+package com.sap.cloud.lm.sl.cf.process.util;
+
+import java.nio.file.Path;
+
+import com.sap.cloud.lm.sl.common.SLException;
+
+public interface ExtractStatusCallback {
```

Comment:
Why use a callback at all? Why not just give the file size limit to ApplicationArchiveExtractor and let him do the check?

Reply:
If the file size limit is to be passed to ApplicationArchiveExtractor, it'd mean it has to be aborted (the extraction) and an exception is to be thrown, then the methods will obviously return null and any track of the file will be lost, so cleanup will not be possible. One way to handle this is to cleanup when throwing the exception (in ApplicationArchiveExtractor) but that breaks clean code standards as all exceptions are to be handled in a central place (UploadAppStep). So Callback is used to track the file on creation, and when an error occurs the file is cleaned up. It also is consistent with uploading to the controller (which also uses a Callback). Both callbacks are in UploadAppStep, one for extraction, other for uploading, both handling errors.

Reply:
I'm not sure it would break clean code standards, as the exception itself is handled in UploadAppStep. IMO, the clean-up of a resource, which is created by class A should be the responsibility of class A, at least in case of an error. And even if this does break clean code standards - so what? Isn't it more complex to have a callback infrastructure + an implementation instead of a simple "if" in ApplicationArchiveExtractor with the corresponding clean up? How is the clean code variant "cleaner"? It is more flexible - that's for sure, but do we need this flexibility?